packages/lib/services/synchronizer/syncInfoUtils.ts

```diff
@@ -13,6 +13,44 @@ const fastDeepEqual = require('fast-deep-equal');
 
 const logger = Logger.create('syncInfoUtils');
 
+type MasterKey = {
+	created_time: number;
+	encryption_method: number;
+	hasBeenUsed: boolean;
+	id: string;
+	source_application: string;
+	updated_time: number;
+};
+
+type PPK = {
+	createdTime: number;
+	id: string;
+	keySize: number;
+	privateKey: {
+		ciphertext: string;
+		encryptionMethod: number;
+	};
+	publicKey: string;
+};
+
+type SyncInfoParsed = {
+	activeMasterKeyId: {
+		updatedTime: number;
+		value: string;
+	};
+	appMinVersion: string;
+	e2ee: {
+		updatedTime: number;
+		value: boolean;
+	};
+	masterKeys: MasterKey[];
+	ppk: {
+		updatedTime: number;
+		value: PPK;
+	};
+	version: number;
+};
```

Comment:
As discussed, all this needs to be removed since we obviously can't have the sync info being defined twice.

When JSON parsing fails, isn't it possible to set this info to an empty value and let the class initialise it automatically?

Reply:
@pedr, not sure if you saw this comment. The SyncInfo should not be defined twice (it's already defined as class). And same comment - why is it not possible to set it to an empty value somewhere and let the existing code set the default values?

Reply:
Ok I think I understand now. You've created this type because previously `s` was `any`, is that right? Indeed we should probably remove `any`. However in this case we don't know the type. It's probably the sync info, or it could be a random string, or an empty string. What if you set the type to `unknown` - would that solve the issue without having this duplicate type information?

Reply:
`unknown` doesn't work because it will assign unknown to all values that we derive from `s`, which is incompatible with `SyncInfo`. I think if we don't want to duplicate the typing information we should keep using `any` since it is the more permissive type.

Reply:
Then use type assertion?